src/TodoApp/Request.hs

```diff
@@ -52,95 +55,117 @@ data Request r where
-  res <- request R.NoReqBody R.GET path [] host
+  res <- request R.NoReqBody R.GET path [] conn
   let v = fromJSON $ fromMaybe (object []) $ decode res
   pure $ case v of
     Aeson.Success a -> a
     Aeson.Error e -> error e
 
 -- | Return pending items in the table
-view :: URI -> IO [Task]
-view host = do
+view :: Connection -> IO [Task]
+view conn = do
   path <- traverse mkPathPiece ["todos"]
   q <- QueryParam <$> mkQueryKey "done" <*> mkQueryValue "is.false"
-  res <- request R.NoReqBody R.GET path [q] host
+  res <- request R.NoReqBody R.GET path [q] conn
   let v = fromJSON $ fromMaybe (object []) $ decode res
   pure $ case v of
     Aeson.Success a -> a
     Aeson.Error e -> error e
 
 -- | Add a new task to the table
-add :: String -> URI -> IO ()
-add task host = do
+add :: String -> Connection -> IO ()
+add task conn = do
   let payload =
         object ["task" .= task]
   path <- traverse mkPathPiece ["todos"]
-  void $ request (R.ReqBodyJson payload) R.POST path [] host
+  void $ request (R.ReqBodyJson payload) R.POST path [] conn
 
 -- | Delete a TODO item with given id
-delete :: Int -> URI -> IO ()
-delete id host = do
+delete :: Int -> Connection -> IO ()
+delete id conn = do
   path <- traverse mkPathPiece ["todos"]
   q <- QueryParam <$> mkQueryKey "id" <*> mkQueryValue ("eq." <> pack (show id))
-  void $ request R.NoReqBody R.DELETE path [q] host
+  void $ request R.NoReqBody R.DELETE path [q] conn
 
 -- | Remove all the TODO items from the table
-reset :: URI -> IO ()
-reset host = do
+reset :: Connection -> IO ()
+reset conn = do
   path <- traverse mkPathPiece ["todos"]
-  _ <- request R.NoReqBody R.DELETE path [] host
+  _ <- request R.NoReqBody R.DELETE path [] conn
   -- Call a SQL function that sets the sequence to start from 1
   path <- traverse mkPathPiece ["rpc", "reset_id"]
-  void $ request R.NoReqBody R.POST path [] host
+  void $ request R.NoReqBody R.POST path [] conn
 
 -- TODO: Add more comments
 
 -- | Make a http request to postgrest service
 request ::
   ( R.HttpBodyAllowed (R.AllowsBody method) (R.ProvidesBody body),
-    MonadIO m,
     R.HttpMethod method,
     R.HttpBody body
   ) =>
   body ->
   method ->
   [RText 'PathPiece] ->
   [QueryParam] ->
-  URI ->
-  m ByteString
-request body method paths qs postgrestUri =
-  R.runReq R.defaultHttpConfig $ do
-    let uri =
-          postgrestUri
+  Connection ->
+  IO ByteString
+request body method paths qs conn = do
+  manager <- case conn of
+        TCP _ -> pure R.defaultHttpConfig
+        UnixSocket socketPath -> do
+          socketManager <- createUnixSocketManager socketPath
+          pure $ R.defaultHttpConfig { R.httpConfigAltManager = Just socketManager }
+  R.runReq manager $ do
+    let uri = case conn of
+          TCP u -> u
+          -- Even though the connection is over unix socket, `req` expects a URL-like string,
+          -- so we use a dummy hostname.
+          UnixSocket _ -> fromJust $ mkURI "http://localhost"
```

Comment:
Move this to a separate function that takes `Connection` as argument and returns the URI.

Reply:
https://github.com/juspay/todo-app/pull/20/commits/5bae8175b4d014173876fd4fd45da59024b3bb77